bindings/matrix-sdk-ffi/src/room_list.rs

```diff
@@ -584,16 +584,32 @@ impl RoomListItem {
     }
 
     /// Builds a `Room` FFI from an invited room without initializing its
-    /// internal timeline
+    /// internal timeline.
     ///
-    /// An error will be returned if the room is a state different than invited
+    /// An error will be returned if the room is a state different than invited.
     ///
     /// ⚠️ Holding on to this room instance after it has been joined is not
-    /// safe. Use `full_room` instead
+    /// safe. Use `full_room` instead.
+    #[deprecated(note = "Please use `room_without_timeline` instead.")]
     fn invited_room(&self) -> Result<Arc<Room>, RoomListError> {
-        if !matches!(self.membership(), Membership::Invited) {
+        self.room_without_timeline()
+    }
+
+    /// Builds a `Room` FFI from a room without initializing its internal
+    /// timeline.
+    ///
+    /// An error will be returned if the room is a state other than invited
+    /// or knocked, the 2 states that would match this use case.
+    ///
+    /// ⚠️ Holding on to this room instance after it has been joined is not
+    /// safe. Use `full_room` instead.
```

Comment:
What do you mean by "not safe" here? This has quite a precise meaning in Rust, and I'm not sure if it's that, and I'm curious :)

Reply:
This is copied from the `invited_room` fn, so we'd probably better ask @stefanceriu , but I think it means 'not safe' as in 'anything related to the timeline will fail'.

Reply:
Resolving this as the doc comment doesn't have the 'not safe' part anymore.